### DISCUSSION TITLE
New features and stability 🎉

### DISCUSSION THREAD

**Discussion by assafelovic** [assafelovic/gpt-researcher#321] [Announcements]
Excited to kick off the new year with a long awaited feature: Research report on specific urls! 🎉 

You can now skip the search by providing urls directly to GPTResearcher and create a research report like so:
```
urls = ["https://docs.tavily.com/docs/tavily-api/introduction",
        "https://docs.tavily.com/docs/tavily-api/python-sdk",
        "https://docs.tavily.com/docs/tavily-api/rest_api"]

query = "How can I integrate Tavily Rest API with my application?"

async def get_report(query: str, source_urls: list) -> str:
    researcher = GPTResearcher(query=query, source_urls=source_urls)
    report = await researcher.run()
    return report

report = asyncio.run(get_report(query, urls))
print(report)
```

The release includes additional stability and performance improvements, along with updated library dependencies.

<hr /><em>This discussion was created from the release <a href='https://github.com/assafelovic/gpt-researcher/releases/tag/v0.1.2'>New features and stability 🎉</a>.</em>

**Comment by aallsbury** [comment]
Could this be tweaked to supply txt or pdf files from local?

**Reply by assafelovic** [reply]
@aallsbury we’re releasing support for local documents next week! Stay tuned

**Reply by aallsbury** [reply]
Awesome, can't wait! Thanks a ton.
